transport.c

```diff
@@ -880,8 +880,8 @@ int transport_set_option(struct transport *transport,
 	/* If either report is 0, report 0 (success). */
 	if (!git_reports || !protocol_reports)
 		return 0;
-	/* If either reports -1 (invalid value), report -1. */
-	if ((git_reports == -1) || (protocol_reports == -1))
+	/* If either reports 1 (invalid value), report -1. */
```

Comment:
This is not what your patch does, as it leaves the check of `protocol_reports` alone. So it still checks that variable for `-1`!

The rationale of changing the check to `git_reports == 1` might sound sensible, but when you read the comment below ("Otherwise if both report unknown, report unknown."), you will realize that [`set_git_option()`](https://github.com/git/git/blob/5da49b4fd58a8cc3e3429fec64412e57dd4e87ca/transport.c#L125-L176) was intended to return `1` upon unknown option, and `-1` upon invalid value for *known* option.

That is, this change is incorrect, but you could arguable change the `die()` in https://github.com/git/git/blob/5da49b4fd58a8cc3e3429fec64412e57dd4e87ca/transport.c#L153 to a `return error()`.